### PR TITLE
[master] minor updates to add license and parent pom

### PR DESCRIPTION
**Pull request by dazey3**
Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#source files
```
Comment (lukasj) [inline .gitattributes:1]: I do not think we need this file as all we'll have here are java, md, xml and probably few sh files

Comment (dazey3) [reply .gitattributes:1]: fair, enough. I figured it didn't hurt to specify file extensions that don't currently exist, but might in the future. I have no problems reducing it down though to the currently relevant extensions. 

Also, thoughts on `NOTICE.md`? There are a huge list of third-party references that aren't used here

File: NOTICE.md [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+https://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License v1.0
+which is available at https://www.eclipse.org/org/documents/edl-v10.php.
+
+SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+
+## Source Code
+
+The project maintains the following source code repositories:
+
+* https://github.com/eclipse-ee4j/eclipselink-asm
```
Comment (lukasj) [inline NOTICE.md:40]: all our repos should be listed here, see https://www.eclipse.org/projects/tools/documentation.php?id=ee4j.eclipselink which can generate up-to-date the Notice file for you. I suggest following edit to the generated file:

* shorten the list of items in 3rd party content, basically in this repo, we depend on asm only, so I think it should be fine to keep just that one

**Review by lukasj — COMMENTED** [review]
add CONTRIBUTING file - again, https://www.eclipse.org/projects/tools/documentation.php?id=ee4j.eclipselink can help; same comment as the one to NOTICE file applies + replace pointer to bugzilla with the pointer to github issues (either in this project or directly in eclipselink; latter may be better) there

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.DS_Store
+*.class
+*.classpath
+*.settings
+*.project
+target/
+*.iml
+.idea/
+test_*.properties
+.sonar
```
Comment (lukasj) [inline .gitignore:10]: this file can be as simple as one line with `target/`. I'd remove at least `*.class`, `test_*.properties` and `.sonar` from this list..

File: org.eclipse.persistence.asm/src/main/resources/about.html [hunk 4]
```diff
@@ -16,10 +16,10 @@
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"/>
-<title>About ASM ${asm.version}</title>
+<title>About ASM v9.1.0</title>
```
Comment (lukasj) [inline org.eclipse.persistence.asm/src/main/resources/about.html:19]: Is this change intentional? maven can replace the property with the real value for us during the build, so we don't have to worry about keeping the version number in the file up-to-date ourselves

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        <build.qualifier>v${maven.build.timestamp}</build.qualifier>
+        <bundle.version>${project.version}.${build.qualifier}</bundle.version>
+        <project.build.testReports.subdirectory>test-reports</project.build.testReports.subdirectory>
+        <project.build.testReports.summaryFile>test-summary</project.build.testReports.summaryFile>
+
+        <!--Common properties-->
+        <common.licenceFiles>LICENSE.md, NOTICE.md</common.licenceFiles>
+
+        <!--Eclipse Dependencies version-->
+        <eclipselink.asm.version>9.1.0</eclipselink.asm.version>
```
Comment (lukasj) [inline pom.xml:107]: rename this property to `asm.version`

File: NOTICE.md (hunk 2 above)
Comment (lukasj) [reply NOTICE.md:40]: and junit needs to stay in the list of 3rd party libs

File: org.eclipse.persistence.asm/src/main/resources/about.html (hunk 4 above)
Comment (dazey3) [reply org.eclipse.persistence.asm/src/main/resources/about.html:19]: I had made this change because I wasn't seeing the variable get replaced when I ran a build in eclipselink/master, but now it's working so i dont know... I'll put it back

**Review by lukasj — COMMENTED** [review]
the pom from https://github.com/eclipse-ee4j/eclipselink/blob/master/plugins/org.eclipse.persistence.asm/pom.xml should have been taken `as is` with probably one or two line changes (pointers to the repo root folder) there is no reason to change it or to introduce additional parent pom - that just complicates things in this case

File: org.eclipse.persistence.asm/src/main/resources/about.html (hunk 4 above)
Comment (lukasj) [reply org.eclipse.persistence.asm/src/main/resources/about.html:19]: try in eclipselink repo:

```
cd plugins/org.eclipse.persistence.asm
mvn -U -C clean install -Poss-release
```

that's how latest (9.1.0) version was produced. I expect that when I clone this repo and do:

```
cd org.eclipse.persistence.asm
mvn -U -C clean install -Poss-release
```

I get the same result (but for sth like 9.2.0-SNAPSHOT)

File: pom.xml (hunk 5 above)
Comment (dazey3) [reply pom.xml:107]: the `org.eclipse.persistence.asm` module pom declares `asm.version` already:
```xml
        <asm.version>9.1</asm.version>
```
From my understanding, `asm.version` refers to the ASM project version, but `eclipselink.asm.version` refers to this projects version.

File: NOTICE.md (hunk 2 above)
Comment (dazey3) [reply NOTICE.md:40]: I'm not sure what I am supposed to do with https://www.eclipse.org/projects/tools/documentation.php?id=ee4j.eclipselink. I navigate there and select "EclipseLink". Then it redirects me to my user profile... what is it supposed to do?

> keep junit

Which junit are you referring to (https://github.com/eclipse-ee4j/eclipselink/blob/master/NOTICE.md)? `pax-exam-junit4 (4.13.1)`?

Comment (dazey3) [reply NOTICE.md:40]: Ok, I got the tool working in IE. For some reason it doesnt work for me in firefox or chrome... 

> all our repos should be listed here

Ok, Ill put them all back, I just didn't think it made sense for THIS project. But maybe I am not thinking about the correct "project". Does "Project" refer to "eclipselink-asm" or "https://projects.eclipse.org/projects/ee4j.eclipselink"?

Comment (lukasj) [reply NOTICE.md:40]: in this case it refers to ee4j.eclipselink project, the owner of the eclipselink-asm (and others) repo(s)

wrt junit, I ment junit:junit:4.13.2 - the one which eclipselink-asm lib has defined in its pom/dependencies

File: pom.xml (hunk 5 above)
Comment (lukasj) [reply pom.xml:107]: there is no relationship between these two pom files

File: org.eclipse.persistence.asm/src/main/resources/about.html (hunk 4 above)
Comment (dazey3) [reply org.eclipse.persistence.asm/src/main/resources/about.html:19]: > mvn -U -C clean install -Poss-release
> [WARNING] The requested profile "oss-release" could not be activated because it does not exist.
I can see no reference to this profile in eclipselink/master/plugins/org.eclipse.persistence.asm. Is this right?

My validation is that I execute `mvn -U -C clean install` and then run a compare against https://mvnrepository.com/artifact/org.eclipse.persistence/org.eclipse.persistence.asm/9.1.0

File: pom.xml (hunk 5 above)
Comment (dazey3) [reply pom.xml:107]: Fine, Ill just get rid of the parent pom that you don't want

File: NOTICE.md (hunk 2 above)
Comment (dazey3) [reply NOTICE.md:44]: > I ment junit:junit:4.13.2

Interesting, I didn't see this in https://www.eclipse.org/projects/tools/documentation.php?id=ee4j.eclipselink generated NOTICE.md. How would you like it to look?

Comment (dazey3) [reply NOTICE.md:44]: I added 
```
JUnit (4.13.2)

* License: Eclipse Public License
```
Hopefully, this works

Comment (lukasj) [reply NOTICE.md:44]: yes, I'll have to revisit this one after waking up. Too late here already ;-)

File: org.eclipse.persistence.asm/pom.xml [hunk 6]
```diff
@@ -18,23 +18,16 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
     <modelVersion>4.0.0</modelVersion>
 
-    <parent>
-        <groupId>org.eclipse.ee4j</groupId>
-        <artifactId>project</artifactId>
-        <version>1.0.6</version>
-        <relativePath/>
-    </parent>
```
Comment (lukasj) [inline org.eclipse.persistence.asm/pom.xml:26]: this parent definition needs to stay - it defines profiles for deployment (ie `oss-release`) among few other important things

compare running `mvn help:effective-pom` without and with this parent defined (`mvn -Poss-release -Dgpg.skip=true ...` does not work; note that `-Dgpg.skip=true` is needed to bypass signing built artifacts by gpg, required unless one has gpg configured on his box)

Comment (dazey3) [reply org.eclipse.persistence.asm/pom.xml:26]: > compare running mvn help:effective-pom without and with this parent defined

Is this pom used to upload to maven central? I can see that it adds sonatype distribution information. I guess this is just a requirement of https://github.com/eclipse-ee4j. I'll add it back

Comment (lukasj) [reply org.eclipse.persistence.asm/pom.xml]: > Is this pom used to upload to maven central? 

yes

>  I guess this is just a requirement[..]

No. One can use whatever he wants as long as the project gets deployed through jakarta.oss.sonatype.org repo. This common parent just makes that easier and brings in common setup and same user/dev experience to projects using it when working with other projects; ie I'm not aware of a ee4j project which does not use it

**Review by lukasj — APPROVED** [review]
I think this is good enough and ready for merge. The are few minor things in notice (libraries listed) and contributing (pointer to issue tracker) files which I'll try to resolve in an extra PR

**Comment by dazey3** [comment]
@lukasj 
Thank you for assisting with a review.

> The are few minor things in notice (libraries listed) and contributing (pointer to issue tracker) files which I'll try to resolve in an extra PR

I'm not sure what issues you see that you want resolved. I can make those changes here to save you the trouble if you want to post them here

**Comment by lukasj** [comment]
wrt notice file: it should list following dependencies:

```xml
            <dependency>
                <groupId>org.ow2.asm</groupId>
                <artifactId>asm</artifactId>
                <version>${asm.version}</version>
            </dependency>
            <dependency>
                <groupId>org.ow2.asm</groupId>
                <artifactId>asm-commons</artifactId>
                <version>${asm.version}</version>
            </dependency>
            <dependency>
                <groupId>org.ow2.asm</groupId>
                <artifactId>asm-tree</artifactId>
                <version>${asm.version}</version>
            </dependency>
            <dependency>
                <groupId>junit</groupId>
                <artifactId>junit</artifactId>
                <version>${junit.version}</version>
            </dependency>
``` 

for asm 9.1, we do have CQ 23221; we should file new one for version 9.2
for junit 4.13.2 I've just filed CQ 23636 (just to be sure)

wrt contributing file - we should add a link to github issue tracker and say sth like that we prefer new issues to be filed there and bugzilla should be used just for checking existing ones - no new issues should be filed there. Our preference going forward is the github issue tracker.

**Comment by lukasj** [comment]
(and project version should be set to 9.1.1-SNAPSHOT; once we integrate asm 9.2, then it should switch to 9.2.0-SNAPSHOT)

**Comment by rfelcman** [comment]
To @dazey3 
Hi Will, do You want help with pr-verification, nighty build and release jobs?
e.g. https://github.com/eclipse-ee4j/eclipselink/blob/master/etc/jenkins/build_asm.groovy
https://github.com/eclipse-ee4j/eclipselink/blob/master/etc/jenkins/release_asm.groovy

**Comment by dazey3** [comment]
@lukasj 

> wrt notice file: it should list following dependencies:

I'm not sure what you think this should look like. Currently, NOTICE.md lists only the "ASM" and "JUnit" artifacts and doesnt make a distinction between different artifacts for ASM (asm, asm-commons, asm-tree). Do we really need separate license declarations for each bundle? It would be like if someone using EclipseLink needed separate license declarations for org.eclipse.persistence.core, org.eclipse.persistence.jpa, org.eclipse.persistence.jpql, org.eclipse.persistence.moxy, ect. It seems to my laymen perspective that all of the ASM artifacts (asm, asm-commons, asm-tree) are just under the one ASM license (https://asm.ow2.io/license.html)

> wrt contributing file - we should add a link to github issue tracker and say sth like that we prefer new issues to be filed there

Are you going to update the documentation tool (https://www.eclipse.org/projects/tools/documentation.php) to reflect that change as well?

**Comment by dazey3** [comment]
@lukasj 

I made the changes for "9.1.0-SNAPSHOT" -> "9.1.1-SNAPSHOT" and to add a link to github issue tracker. I'm not sure what you'd like the separate ASM artifact license declarations to look like specifically.

**Comment by dazey3** [comment]
> do You want help with pr-verification, nighty build and release jobs? e.g. https://github.com/eclipse-ee4j/eclipselink/blob/master/etc/jenkins/build_asm.groovy https://github.com/eclipse-ee4j/eclipselink/blob/master/etc/jenkins/release_asm.groovy

I see. Are these groovy files how EclipseLink automatically runs and pushes nightly builds (https://www.eclipse.org/eclipselink/downloads/nightly.php)?  

I honestly don't know how the build infrastructure works for Eclipse builds. I would appreciate any help in getting this stuff setup. I'm looking in `https://github.com/eclipse-ee4j/eclipselink/blob/master/etc` and I don't know what is/isn't important.

**Comment by lukasj** [comment]
> Currently, NOTICE.md lists only the "ASM" and "JUnit" artifacts and doesnt make a distinction between different artifacts for ASM (asm, asm-commons, asm-tree). Do we really need separate license declarations for each bundle? 

I don't know. For spring, logback and few others, there is one entry per jar 

> It would be like if someone using EclipseLink needed separate license declarations for org.eclipse.persistence.core, org.eclipse.persistence.jpa, org.eclipse.persistence.jpql, org.eclipse.persistence.moxy, ect. It seems to my laymen perspective that all of the ASM artifacts (asm, asm-commons, asm-tree) are just under the one ASM license (https://asm.ow2.io/license.html)

(IANAL) There is no rule saying that all artifacts originating from one project must have exactly same license as long as those licenses are compatible; there are cases where one jar is under say EDL and other under EPL. 


> Are you going to update the documentation tool (https://www.eclipse.org/projects/tools/documentation.php) to reflect that change as well?

No, I can't do that. Filed https://bugs.eclipse.org/bugs/show_bug.cgi?id=575198
